packages/plugin-ext/src/main/node/handlers/plugin-vscode-file-handler.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { PluginDeployerFileHandler, PluginDeployerEntry, PluginDeployerFileHandlerContext } from "../../../common/plugin-protocol";
+import { injectable } from "inversify";
+import * as os from "os";
+import * as path from "path";
+
+@injectable()
+export class PluginVsCodeFileHandler implements PluginDeployerFileHandler {
+
+    private unpackedFolder: string;
+    constructor() {
+        this.unpackedFolder = path.resolve(os.tmpdir(), 'vscode-unpacked');
+    }
+
+    accept(resolvedPlugin: PluginDeployerEntry): boolean {
+        return resolvedPlugin.isFile() && resolvedPlugin.path() !== null && resolvedPlugin.path().endsWith(".vsix");
+    }
+
+    async handle(context: PluginDeployerFileHandlerContext): Promise<void> {
```

Comment:
Maybe it worsts to move into abstract class, code is identical as in `PluginTheiaFileHandler`? WDYT?

Reply:
I will move vscode in another npm module so probably abstract class may be less useful afterwards

Reply:
ok